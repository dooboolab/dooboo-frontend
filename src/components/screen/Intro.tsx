import React, { Component } from 'react';
import { Redirect } from 'react-router-dom';

import Button from '../shared/Button';

import classNames from 'classnames/bind';
const classes = require('./Intro.css');
const cx = classNames.bind(classes);

class Intro extends Component<any, any> {
  public render() {
    const containerClass = cx({
      container: true,
      background: true,
    });
    return(
      <div className={containerClass}>
      <div className={classes.box}>
          <Button
            id='btn'
            onClick={() => this.onClick()}
            white={true}
            txt='Button'
          />
        </div>
      </div>
    );
  }

  private onClick = () => {
    alert('clicked');
  }
}

export default Intro;
