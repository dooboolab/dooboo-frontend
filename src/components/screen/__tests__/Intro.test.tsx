import React from 'react';
import renderer from 'react-test-renderer';

import Intro from '../Intro';

// test for the container page in dom
describe('Intro page DOM rendering test', () => {
  let tree;
  const component = <Intro />;

  it('component and snapshot matches', () => {
    tree = renderer.create(component).toJSON();
    expect(tree).toMatchSnapshot();
  });
});

describe('Interaction', () => {
  let rendered;
  let instance;
  const component = <Intro/>;

  beforeAll(() => {
    rendered = renderer.create(component);
    instance = rendered.root;

    Object.defineProperty(window.alert, 'window.alert', {
      configurable: true,
    });
    window.alert = jest.fn();
  });

  it('Simulate onClick', () => {
    const spy = jest.spyOn(rendered.getInstance(), 'onClick');
    const button = instance.find(
      (el: any) => el.props.id === 'btn',
    );
    button.props.onClick();
    expect(spy).toBeCalled();
  });
});
