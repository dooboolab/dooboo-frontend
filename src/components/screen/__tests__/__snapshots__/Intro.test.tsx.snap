// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Intro page DOM rendering test component and snapshot matches 1`] = `
<div
  className="container background"
>
  <div
    className="box"
  >
    <div
      className="btn"
      onClick={[Function]}
      style={
        Object {
          "borderColor": "#fff",
          "color": "#fff",
        }
      }
    >
      <span
        className="txt"
      >
        Button
      </span>
    </div>
  </div>
</div>
`;
